beacon-chain/rpc/over/handlers_deposit.go

```diff
@@ -271,7 +333,7 @@ func buildPendingDepositEstimations(st state.BeaconState, pubkey []byte, initial
 
 		// If the pending deposit has same pubkey as the one we are looking for
 		// append it to the response
-		if bytes.Equal(pd.PublicKey, pubkey) {
+		if pubkeyFilter != nil && bytes.Equal(pd.PublicKey, pubkeyFilter) {
```

Comment:
If `pubkeyFilter == nil`, there will be no append operation on `pdes`, so it will be an empty list(`make([]*structs.PendingDepositEstimationContainer, 0)`).